api/v1alpha3/awsmachine_types.go

```diff
@@ -36,6 +36,11 @@ type AWSMachineSpec struct {
 	// AMI is the reference to the AMI from which to create the machine instance.
 	AMI AWSResourceReference `json:"ami,omitempty"`
 
+	// ImageLookupFormat is the format string used for image lookup if AMI is
+	// not set. Supports substitutions for $BASE_OS and $K8S_VERSION.
```

Comment:
```suggestion
	// not set. Supports substitutions for ${BASE_OS} and ${K8S_VERSION}.
```

Reply:
I originally substituted the variables in this way, however it causes the `clusterawsadm` installation to bomb, because it expects an environment variable to be passed in to the manifest. This is only a token for string substitution.

Reply:
These two comments are that the actual string substitution doesn't match the docs. I think if a user supplies `imageLookupFormat: '$BASE_OS-$K8S_VERSION'` they'll have a much worse time than if they'd said `imageLookupFormat: '${BASE_OS}-${K8S_VERSION}'`

Reply:
Perhaps I need to use a different type of substitution? '{{ BASE_OS }}-{{ K8S_VERSION }}'? That way the docs can match without overhauling the token parser in `clusterctl`

Reply:
Oh, I see what you mean now – yeah, I think a different token would be preferable then anyway. Maybe consider a simple https://golang.org/pkg/text/template/ ?

Reply:
Thank you for the suggestion. Using golang templates.